examples/demo.rs

```diff
@@ -58,11 +62,23 @@ fn main() -> Result<()> {
 
     let completer = Box::new(DefaultCompleter::new_with_wordlen(commands.clone(), 2));
 
+    let cursor_shapes = HashMap::from([
+        (
+            PromptEditMode::Vi(reedline::PromptViMode::Insert),
+            CursorShape::Line,
+        ),
+        (
+            PromptEditMode::Vi(reedline::PromptViMode::Normal),
+            CursorShape::Block,
+        ),
```

Comment:
I'm thinking we should add emacs mode in here too, so people can set their cursor shape when not using Vi mode. Thoughts?

Reply:
I guess my vote is to allow the cursor to be set in all modes, insert, normal, emacs (i think that's all modes), via a configuration setting (in nushell) and when it's not set specifically, it defaults to "none" so that ansi escape sequences work.